api/features_api.py

```diff
@@ -375,17 +376,19 @@ def do_patch(self, **kwargs):
 
     return {'message': f'Feature {feature_id} updated.'}
 
-  @permissions.require_create_feature
-  def do_delete(self, **kwargs) -> dict[str, str]:
+  def do_delete(self, **kwargs) -> flask.Response | dict[str, str]:
     """Delete the specified feature."""
     # TODO(jrobbins): implement undelete UI.  For now, use cloud console.
-    feature_id = kwargs.get('feature_id', None)
+    if 'feature_id' not in kwargs:
+      self.abort(404, msg='Feature ID not specified')
+    feature_id: int = int(kwargs.get('feature_id', 0))
+    # Validate the user has edit permissions and redirect if needed.
+    redirect_resp = permissions.validate_feature_edit_permission(
+        self, feature_id)
+    if redirect_resp:
+      return redirect_resp
```

Comment:
I'm realizing after seeing this that there is no redirect that can occur with an API request like this, so this will not behave the expected way (i.e. redirecting the user somewhere else). Instead we should just have an abort with a reason.

```suggestion
    # Validate the user has edit permissions and redirect if needed.
    redirect_resp = permissions.validate_feature_edit_permission(
        self, feature_id)
    if redirect_resp:
      self.abort(403, msg='User does not have permission to edit this feature.')
```

I know many of the endpoints we have are doing this incorrectly, so that's something to take care of in the future. 😅